azure-pipelines.yml

```diff
@@ -1,15 +1,37 @@
+variables:
+  CI_COMMON_VERSION: "1.0"
 
-jobs:
-  - job: build_docker_images
-    displayName: Build Docker Images
-    condition: not(eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+stages:
+- stage: build_common_docker_images
+  displayName: Build Common Docker Images
+  condition: always()
+  jobs:
+  - job:
+    condition: eq(variables['Build.SourceBranch'], 'refs/heads/master')
```

Comment:
So this job still gets triggered on any commit to master? Is there a way of committing a change to master that triggers all _but_ the ci-common image being built (in the more common case where you're only changing VFX dependencies)?

Reply:
I would also like another way of triggering these, a simple update to the readme will consume many hours of compute, and that makes me feel bad for the planet!
I'm thinking of using tags to trigger the pushes, and more fine grain triggers to only run relevant jobs. But that will take some more work.

Reply:
I think part of this can be done with path filters:

https://docs.microsoft.com/en-us/azure/devops/pipelines/build/triggers?view=azure-devops&tabs=yaml#paths

We can always exclude certain directories or files in the repo from the triggers.

For the other aspect of this, perhaps we could have a refs/head/common branch and a refs/head/master branch and trigger ci-common from the former and all the others from the latter?

I think I'd prefer triggering from branches than tags to be honest, just because it would be a simpler workflow, but I appreciate the desire to use tags. It's complicated further by having Git tags and Docker tags which wouldn't necessarily line up too.

Reply:
Having multiple branches sounds like a good safety. You would still want a commit to refs/head/common to trigger a push of refs/head/master in that case, but that would occur much less often.